File: PieGallery/app/src/main/java/com/deadely/piegallery/base/BasePresenter.java
package com.deadely.piegallery.base;

import android.os.Bundle;

import org.jetbrains.annotations.NotNull;

public abstract class BasePresenter<T extends IBaseMvpView> implements IBaseMvpPresenter<T> {

    protected T mMvpView;

    @Override
    public void attachView(T mvpView) {
        mMvpView = mvpView;
    }

    @Override
    public void detachView() {
        mMvpView = null;
    }

    public T getMvpView() {
        return mMvpView;
    }

    @Override
    public void onCreate(@NotNull Bundle savedInstanceState) {
        return;
    }

    @Override
    public void setArguments(@NotNull Object... objects) {
        return;
    }
}

File: PieGallery/app/src/main/java/com/deadely/piegallery/base/BasePresenter.kt
package com.deadely.piegallery.base

import android.content.Context
import com.github.terrakok.cicerone.Router
import dagger.hilt.android.qualifiers.ApplicationContext
import moxy.MvpPresenter
import moxy.MvpView
import javax.inject.Inject

abstract class BasePresenter<T : MvpView> : MvpPresenter<T>() {

    @Inject
    lateinit var router: Router

}
